### PR TITLE
Replace Feedback Fish script with custom modal

### PR DESCRIPTION
**Pull request by delucis**
#### What kind of changes does this PR include?

- Changes to the docs site code

#### Description

- Replaces Feedback Fish’s default script with an Astro component that shows a similar modal instead.
- Slightly less fancy than the current modal but much cheaper in terms of JS etc. (like 150KiB cheaper 😅)

| Before | After |
|---|---|
| image | image |
| image | image |

### CODE REVIEW

**Comment by Yan-Thomas** [comment]
Thanks, @delucis. The i18n, performance, and design cohesion improvements here make this custom component such a nice addition! I'll point out that we don't clear the text area after you send the form in case this was something you didn't notice rather than intentional behavior.

**Comment by Jutanium** [comment]
This looks fantastic! Nice Web Component!

**Review by jasikpark — COMMENTED** [review]
lgtm

**Comment by tony-sull** [comment]
150kb 👀 

The `dialog` element is (sadly) not all that [useful today](https://a11y-dialog.netlify.app/further-reading/dialog-element).  [a11y-dialog](https://a11y-dialog.netlify.app/) is worth a look, that's been my go-to for accessible dialogs & modals with minimal JS

**Comment by MoustaphaDev** [comment]
Awesome work on this @delucis! One thing that bothered me was the color of the `textarea` in darkmode.
For comparison with the searchbox:

Untitled design

I lightmode, the themes of the feebackfish and the Docsearch widget match, not in darkmode.

**Review by sarah11918 — APPROVED** [review]
Approved by Sarah's lab of evil devices. No issues in leaving feedback. Confirmation is adorbs.

File: src/components/tutorial/FeedbackButton.astro [hunk 1]
```diff
@@ -1,44 +1,323 @@
...
+
+	textarea {
+		border: 2px solid var(--theme-dim-lighter);
+		border-radius: 0.5rem;
+		resize: none;
+		line-height: 1.5;
+		padding: 0.5rem;
+		font-family: inherit;
+		font-size: var(--theme-text-xs);
+	}
```
Comment (delucis) [inline src/components/tutorial/FeedbackButton.astro:282]: This change should address @MoustaphaDev's review hopefully!

```suggestion
	}
	
	:global(.theme-dark) textarea {
		background-color: var(--theme-bg-offset);
	}
```

**Review by MoustaphaDev — APPROVED** [review]

**Pull request merged**